ui/imports/shared/views/chat/ImageContextMenu.qml

```diff
@@ -1,29 +1,38 @@
+import QtQuick 2.15
 import StatusQ.Popups 0.1
-
 import utils 1.0
 
 StatusMenu {
     id: root
 
+    property string url
     property string imageSource
 
-    readonly property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
+    QtObject {
+        id: d
+        property bool isUnfurled: (!!url&&url!=="")
+        readonly property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
+    }
 
     StatusAction {
-        text: root.isGif ? qsTr("Copy GIF") : qsTr("Copy image")
+        text: qsTr("Copy ") + (d.isUnfurled ? qsTr("link") :
```

Comment:
Nope, pls use separate texts and don't glue unrelated things together (https://hackmd.io/@status-desktop/HkSPVn9oq#Dynamic)